### ISSUE TITLE
validation(workspace/tune): Phase 2 — UiSchema integration after lizyml ships parameter_bounds() (depends on lizyml#152)

### ISSUE THREAD

**Issue by nbx-liz** [nbx-liz/LizyStudio#461]
## Context

This is the **Phase 2** follow-up to Issue #460 (Phase 1: Studio-side UI guardrails with hardcoded bounds). After lizyml ships the upstream validation work (lizyml issue #152), LizyStudio swaps the temporary hardcoded bounds for the lizyml-provided values and removes any compensating workaround.

This issue is **blocked** until lizyml#152 lands and a release with the new API is published.

---

## Dependency

- **lizyml#152** must merge and ship a release. Required APIs:
  - `EstimatorProvider.parameter_bounds(task: TaskType) -> dict[str, dict[str, float | int]]`
  - `parse_space()` rejects `low >= high` and `log + low <= 0` with typed errors
  - `_expand_range` honors `min_allowed` / `max_allowed` on `SearchDim`
  - `BoundaryDimStatus.clamped_to_bound: bool`

When that release is available:

1. Bump `pyproject.toml` lizyml pin to the new minor.
2. Run `uv lock` and `pnpm generate:api` (if any schema fields changed via re-export).

---

### 1. `src/lizystudio/backends/lizyml_ui_schema.py` — expose bounds via UiSchema

```python
# Inside build_ui_schema(...)
from lizyml.estimators.lgbm.provider import LGBMProvider

provider = LGBMProvider()
parameter_bounds = {
    task: provider.parameter_bounds(task)
    for task in (\"binary\", \"regression\", \"multiclass\")
}
return {
    # ... existing fields ...
    \"parameter_bounds\": parameter_bounds,
}
```

Add `parameter_bounds` to `api/models.py::UiSchemaResponse` so it surfaces in the OpenAPI spec, then `pnpm generate:api`.

### 2. `frontend/src/components/workspace/SearchSpaceRow.tsx` — read from UiSchema

Replace:

```ts
import { PARAMETER_BOUNDS_PHASE1 } from \"./parameter-bounds\";
const bounds = PARAMETER_BOUNDS_PHASE1[param.key];
```

With:

```ts
const bounds = uiSchema?.parameter_bounds?.[task]?.[param.key];
```

(Use the `task` already passed via props.)

### 3. Delete `frontend/src/components/workspace/parameter-bounds.ts`

The temporary map file from Phase 1. Confirmed safe to delete once Step 2 lands.

### 4. `src/lizystudio/backends/lizyml/config_compat.py` — `validate_config` reuses lizyml's enriched parse_space

Today (Phase 1) — `validate_config` only runs `LizyMLConfig.model_validate()` + custom task/objective compat checks. It does NOT exercise `parse_space` directly, so inverted-range / log-non-positive payloads slip through to runtime.

Phase 2 change — call `lizyml.tuning.search_space.parse_space(config[\"tuning\"][\"optuna\"][\"space\"])` inside `validate_config` so the new typed errors from lizyml#152 surface as 400 errors with the same Pydantic-style shape:

```python
from lizyml.tuning.search_space import parse_space
from lizyml.core.exceptions import LizyMLError

def task_params_compat_errors(config: dict) -> list[dict]:
    errors = []
    space = config.get(\"tuning\", {}).get(\"optuna\", {}).get(\"space\", {})
    if space:
        try:
            parse_space(space)
        except LizyMLError as exc:
            errors.append({
                \"loc\": (\"tuning\", \"optuna\", \"space\"),
                \"msg\": exc.user_message,
                \"type\": \"value_error\",
                \"ctx\": exc.context,
            })
    # ... existing compat checks ...
    return errors
```

### 5. (Conditional) Remove any Re-tune adapter post-clamp workaround

If Phase 1 PR included a Studio-side post-clamp on `expand_dims` output (Layer 5 in original Issue #460), remove it now — lizyml owns the proper fix per lizyml#152.

Audit `src/lizystudio/backends/lizyml/training_mixin.py` (or equivalent) for any `_clamp_space_to_bounds` helper added during Phase 1 and delete.

### 6. `BoundaryDimStatus.clamped_to_bound` UI surface

In `frontend/src/components/retune/SearchSpaceEvolutionPanel.tsx` or `BoundaryExpansionPanel.tsx`, render a small badge on dims where `clamped_to_bound === true`:

```tsx
{dim.clamped_to_bound && (
  <Badge variant=\"secondary\" className=\"text-[10px]\">
    bounded
  </Badge>
)}
```

with a tooltip: \"This dimension hit the safe parameter bound and was not expanded further.\"

---

## Tests

- **Backend**: `tests/test_backends_lizyml.py::test_validate_config_rejects_inverted_range` — should be a 400-style Pydantic error after Phase 2 (was unguarded before).
- **Backend**: `test_validate_config_rejects_log_with_nonpositive_low`.
- **Backend**: `test_ui_schema_includes_parameter_bounds_per_task` — assert UiSchema response contains `parameter_bounds[\"binary\"][\"learning_rate\"][\"max\"] == 1.0`.
- **Frontend Vitest**: `SearchSpaceRow.test.tsx` — when the test injects `uiSchema.parameter_bounds.binary.learning_rate.max = 1.0`, typing 5.0 in the Range Max input clamps to 1.0.
- **Frontend Vitest**: `BoundaryExpansionPanel.test.tsx` — `clamped_to_bound: true` renders the \"bounded\" badge.
- **Cleanup verification**: `parameter-bounds.ts` no longer exists (`pnpm test` + `tsc` would fail if anything still imports it).

---

## Acceptance Criteria

- [ ] `pyproject.toml` lizyml pin bumped to the version that includes lizyml#152.
- [ ] `UiSchemaResponse.parameter_bounds` populated per task from `LGBMProvider.parameter_bounds()`.
- [ ] `frontend/src/components/workspace/parameter-bounds.ts` deleted; `SearchSpaceRow.tsx` reads from `uiSchema.parameter_bounds`.
- [ ] `validate_config` returns Pydantic-style errors for inverted ranges and log + low<=0 payloads.
- [ ] Any Phase 1 Re-tune post-clamp workaround removed.
- [ ] `BoundaryDimStatus.clamped_to_bound` surfaced as a UI badge in the Re-tune dashboard.
- [ ] HISTORY.md Decision row updated for the lizyml dep bump and `parameter_bounds` integration.
- [ ] `pnpm test`, `pnpm test:e2e`, `uv run pytest`, `pnpm check`, `pnpm build` all green.

## Related

- **Phase 1 issue**: #460 (Studio-side UI guardrails with hardcoded bounds)
- **Upstream issue**: nbx-liz/LizyML#152 (parameter_bounds + parse_space + expand_dims clamp)
- HISTORY.md H-0061 / H-0068 — Re-tune introduction

**Comment by nbx-liz** [comment]
Completed by P-0104 Wave 3.1a (#473 — objective list + parameter_bounds from LGBMProvider, Range Min/Max clamp, objective drift CI) + Wave 3.1b (#476 — option_sets.metric became the {native, feval} shape from metric_choices, model_metric removed, registry list split into option_sets.eval_metric, Custom (slow) badge, clamped_to_bound badge, metric drift CI). The deferred validate_config → parse_space() rewire is tracked separately in #474.
